src/crypto-api.ts

```diff
@@ -496,6 +496,76 @@ export interface CryptoApi {
      * @param version - The backup version to delete.
      */
     deleteKeyBackupVersion(version: string): Promise<void>;
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    //
+    // Dehydrated devices
+    //
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    /**
+     * Returns whether MSC3814 dehydrated devices are supported by the crypto
+     * backend and by the server.
+     *
+     * This should be called before any of the other dehydrated device
+     * functions are called, and if it returns `false`, none of the other
+     * dehydrated device functions should be called.
+     */
+    isDehydrationSupported(): Promise<boolean>;
+
+    /**
+     * Rehydrate the dehydrated device stored on the server.
+     *
+     * Checks if there is a dehydrated device on the server.  If so, rehydrates
+     * the device and processes the to-device events sent to it.  This function
+     * should be called before a dehydrated device is created for the first
+     * time after the client logs in.
+     *
+     * @returns `true` if a dehydrated device was found; otherwise, `false`.
+     */
+    rehydrateDeviceIfAvailable(): Promise<boolean>;
+
+    /**
+     * Creates and uploads a new dehydrated device.
+     *
+     * If now dehydration key is available in secret storage, a new key is
+     * created.  Most applications should call `scheduleDeviceDehydration` so
+     * that the dehydrated device gets replaced periodically with a new one, to
+     * avoid to-device events stacking up on the server.  However, clients that
+     * want to have more control over the dehydration process may use this
+     * function instead.
+     */
+    createAndUploadDehydratedDevice(): Promise<void>;
+
+    /**
+     * Schedule periodic creation of dehydrated devices.
+     *
+     * If the delay is omitted or 0, this function's promise will resolve after
+     * the first dehydrated device is created.
```

Comment:
```suggestion
     * If the delay is omitted or 0, this function's promise will resolve after
     * the first dehydrated device is created. Otherwise, it resolves immediately.
```